src/Client.php

```diff
@@ -111,16 +126,14 @@ public function runRawQuery(string $queryString, $resultsAsArray = false, array
     {
         $request = new Request($this->requestMethod, $this->endpointUrl);
 
-        foreach($this->httpHeaders as $header => $value) {
-            $request = $request->withHeader($header, $value);
-        }
-
         // Convert empty variables array to empty json object
         if (empty($variables)) $variables = (object) null;
         // Set query in the request body
         $bodyArray = ['query' => (string) $queryString, 'variables' => $variables];
         $request = $request->withBody(Utils::streamFor(json_encode($bodyArray)));
 
+        $request = $this->auth->run($request, $this->httpHeaders, $this->options);
```

Comment:
We don't have the `$this->httpOptions` attribute in this class, so I've added `$this->options` to keep parameters.
Removed `$this->httpHeaders` as they aren't needed anymore.

Reply:
I apologize, I misspoke. I meant using `this->httpHeaders` instead of `this->options`. With the current implementation `this->options` is just a subset of `this->httpHeaders` so I don't see why we can't just pass the existing class variable instead of having to create a new one and pass instead.
Please correct me if I'm wrong here.
Also, I don't understand what you mean by "Removed $this->httpHeaders as they aren't needed anymore." I can still see the variable being initalized and used.

Reply:
I mean the attribute `this->httpHeaders` has **only headers** info, but not all options:
```
        $headers = array_merge(
            $authorizationHeaders,
            $httpOptions['headers'] ?? [],
            ['Content-Type' => 'application/json']
        );
        ...
        $this->httpHeaders = $headers;
```
So, we can't use this->httpHeaders attribute. That's why a new one was added.

---
"Also, I don't understand what you mean by "Removed $this->httpHeaders as they aren't needed anymore." I can still see the variable being initalized and used."
Never mind here, I mean I removed second argument from here:
`$this->auth->run($request, $this->httpHeaders, $this->options);`

Reply:
You are right, thanks for clarifying this. I'm good with the changes now.